### PR TITLE
Refactor graph copy and subgraph functions

### LINKED ISSUE #300: Refactor graph copy functions
Look at draft of  `copyGraph` functions in attic/sandbox.jl
The idea is to have only one function to be called by functions such as `buildSubgraphFromLabels`

Will at least look at it for v0.6.0 and decide if it should be added.

### PR DESCRIPTION
**Pull request by Affie**
Close #300
Still to do:
- [x] add tests for `buildSubgraph`
- [x] deprecate `_copyIntoGraph` in favor of `copyGraph`/`deepcopyGraph`/`deepcopyGraph!`
- [x] deprecate `getSubgraph` in favor of `buildSubgraph`
- [x] deprecate `getSubgraphAroundNode` in favor of `buildSubgraph`

### CODE REVIEW

File: src/LightDFG/entities/LightDFG.jl [hunk 1]
```diff
@@ -66,3 +66,14 @@ LightDFG(description::String,
          sessionData::Dict{Symbol, String},
          solverParams::AbstractParams) =
          LightDFG(FactorGraph{Int,DFGVariable,DFGFactor}(), description, userId, robotId, sessionId, userData, robotData, sessionData, Symbol[], solverParams)
+
+
+LightDFG{T,V,F}(description::String,
```
Comment (dehann) [inline src/LightDFG/entities/LightDFG.jl:71]: should this not perhaps be an inner constructor, or does it not matter?

**Comment by dehann** [comment]
I think we can merge this, right?

**Comment by GearsAD** [comment]
LGTM, just one thing - is there any reason (other than the original function signatures) that we combine variables and factors as input to `copyGraph!`, e.g. ` variableFactorLabels::Vector{Symbol};`?

This just results in the lists being split up first thing into the function. If there isn't a reason beyond the initial design, I think we should split it up to make the calls more efficient.

File: src/LightDFG/entities/LightDFG.jl (hunk 1 above)
Comment (Affie) [reply src/LightDFG/entities/LightDFG.jl:71]: https://docs.julialang.org/en/v1/manual/constructors/#man-inner-constructor-methods-1
> It is good practice to provide as few inner constructor methods as possible: only those taking all arguments explicitly and enforcing essential error checking and transformation. Additional convenience constructor methods, supplying default values or auxiliary transformations, should be provided as outer constructors that call the inner constructors to do the heavy lifting. This separation is typically quite natural.

Comment (dehann) [reply src/LightDFG/entities/LightDFG.jl:71]: got it thanks!

**Comment by GearsAD** [comment]
Btw, I meant that we can do that in a subsequent PR - I'll add that. This is good thanks.

**Comment by GearsAD** [comment]
Oh I see you add it to the list already :) If you haven't started, let's save it for another PR

**Comment by Affie** [comment]
Ok, will do separate PR. Just don't start using it in master yet.

**Pull request merged**